tools/internal_ci/linux/aws/grpc_aws_run_remote_test.sh

```diff
@@ -90,26 +97,29 @@ echo $SERVER_HOST_KEY_ENTRY >> ~/.ssh/known_hosts
 echo "Waiting 2m for instance to initialize..."
 sleep 2m
 
-WORKLOAD=grpc_aws_experiment_remote.sh
-REMOTE_SCRIPT_EXITCODE=0
 echo "Copying workspace to remote instance..."
 # use rsync over ssh since it's much faster than scp
 time rsync -e "ssh -i ~/.ssh/temp_client_key" -a github/grpc ubuntu@$IP:~/workspace
 echo "Beginning CI workload..."
 
+# filename of the test script to execute remotely, relative to gRPC repository root
+# use a default value if the env variable is not set
+REMOTE_WORKLOAD_SCRIPT=${REMOTE_WORKLOAD_SCRIPT:-tools/internal_ci/linux/aws/grpc_aws_experiment_remote.sh}
+
 # run remote workload script in the background, with redirected stdout and stderr
 # to avoid problems with ssh session not closing after the remote script finishes
 # but stdout and stderr are still open because the remote has spawned subprocesses
 # that keep stdout and stderr open.
 # * PID of the process that executes the remote script will be stored in aws_build.pid
 # * stderr and stdout will be streamed to aws_build.log
 # * once done, the exitcode of the remote script will be in aws_build.exitcode
-REMOTE_WORKLOAD_COMMAND="nohup bash -c '(bash grpc/tools/internal_ci/linux/aws/${WORKLOAD}; echo \$? >/tmp/aws_build.exitcode) >>/tmp/aws_build.log 2>&1' >/dev/null 2>&1 & echo \$! >/tmp/aws_build.pid"
+REMOTE_WORKLOAD_COMMAND="nohup bash -c '(bash grpc/${REMOTE_WORKLOAD_SCRIPT}; echo \$? >/tmp/aws_build.exitcode) >>/tmp/aws_build.log 2>&1' >/dev/null 2>&1 & echo \$! >/tmp/aws_build.pid"
 
 # the tail command simply streams the contents of aws_build.log as they become available
 # and stops when the remote workload exits (determined based on the PID)
 SSH_COMMAND='uname -a; rm -f /tmp/aws_build.log /tmp/aws_build.exitcode /tmp/aws_build.pid; touch /tmp/aws_build.log; cd ~/workspace; '"${REMOTE_WORKLOAD_COMMAND};"' tail -f /tmp/aws_build.log --pid $(cat /tmp/aws_build.pid); exit $(cat /tmp/aws_build.exitcode)'
 
+REMOTE_SCRIPT_EXITCODE=0
```

Comment:
I think this would be simpler as:
```
time ssh ...
REMOTE_SCRIPT_EXITCODE=$?
```

Reply:
my only issue with your suggestion is that when only using `REMOTE_SCRIPT_EXITCODE=$?` can get easily broken when modifying the script since `REMOTE_SCRIPT_EXITCODE=$?` needs to always immediately follow the ssh command. The original version is more robust since the relationship between the ssh command and  `$?` is clearer (and harder to accidentally get wrong when modifying the script).